source/write/insert.txt

```diff
@@ -0,0 +1,199 @@
+================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code examples, write, save, create
+
+Overview
+--------
+
+In this guide, you can learn how to use {+driver-short+} to add
+documents to a MongoDB collection by performing insert operations.
+
+An insert operation inserts one or more documents into a MongoDB collection.
+You can perform an insert operation by using the ``insert_one()`` or
+``insert_many()`` method.
+
+.. .. tip:: Interactive Lab
+   
+..    This page includes a short interactive lab that demonstrates how to
+..    insert data by using the ``insert_one()`` method. You can complete this
+..    lab directly in your browser window without installing MongoDB or a code editor.
+
+..    To start the lab, click the :guilabel:`Open Interactive Tutorial` button at the
+..    top of the page. To expand the lab to a full-screen format, click the
+..    full-screen button (:guilabel:`⛶`) in the top-right corner of the lab pane.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<pymongo-get-started>` tutorial.
+
+The ``_id`` Field
+-----------------
+
+In a MongoDB collection, each document *must* contain an ``_id`` field
+with a unique field value.
+
+MongoDB allows you to manage this field in two ways:
+
+- You can set this field for each document yourself, ensuring each
+  ``_id`` field value is unique.
+- You can let the driver automatically generate unique ``ObjectId``
+  values for each document ``_id``. If you do not manually set an
+  ``_id`` value for a document, the driver populates the field
+  with an ``ObjectId``.
+
+Unless you can guarantee uniqueness, we recommend
+letting the driver automatically generate ``_id`` values.
+
+.. note::
+
+   Duplicate ``_id`` values violate unique index constraints, which
+   causes the driver to return a ``WriteError`` from
+   ``insert_one()`` or a ``BulkWriteError`` from ``insert_many()``.
+
+To learn more about the ``_id`` field, see the
+:manual:`Unique Indexes </core/index-unique/>` guide in the MongoDB Server manual.
+
+To learn more about document structure and rules, see the
+:manual:`Documents </core/document>` guide in the MongoDB Server manual.
+
+Insert One Document
+-------------------
+
+To add a single document to a MongoDB collection, call the ``insert_one()``
+method and pass the document you want to add.
+
+The following example inserts a document into the ``restaurants`` collection:
+
+.. code-block:: python
+   :copyable: true
+
+   sample_restaurants.restaurants.insert_one({"name" : "Mongo's Burgers"})
+
+.. TODO: To see a full example of using the ``insert_one()`` method to find a single document, see
```

Comment:
Are these meant to be addressed in future PRs?

Reply:
Yes, any TODOs are intended to be addressed in later PRs once we have some of the other pages written